Simplenote/src/main/java/com/automattic/simplenote/utils/NoteUtils.java

```diff
@@ -53,4 +64,48 @@ public static String getWordCount(String content) {
         int words = (content.trim().length() == 0) ? 0 : content.trim().split("([\\W]+)").length;
         return NumberFormat.getInstance().format(words);
     }
+
+    public static void showDialogDeletePermanently(final Activity activity, final Note note) {
+        new AlertDialog.Builder(new ContextThemeWrapper(activity, R.style.Dialog))
+            .setTitle(R.string.delete_dialog_title)
+            .setMessage(R.string.delete_dialog_message)
+            .setNegativeButton(R.string.no, null)
+            .setPositiveButton(
+                R.string.yes,
+                new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int whichButton) {
+                        if (note != null) {
+                            note.delete();
+
+                            // Show empty placeholder for large devices in landscape.
+                            if (activity instanceof NotesActivity) {
+                                ((NotesActivity) activity).showDetailPlaceholder();
+                            // Close editor for small devices and large devices in portrait.
+                            } else if (activity instanceof NoteEditorActivity) {
+                                ((NoteEditorActivity) activity).finish();
```

Comment:
Minor (🔍 ): Consider removing the cast to `NoteEditorActivity` for this case, it is not needed, isn't it?

Reply:
That's correct.  Since `finish` is available to all `Activity` classes, it's not required to cast `NoteEditorActivity` here.  I try to be a little extra verbose with cases like this when we're doing different things based on layout.  It's not always apparent that Simplenote uses `NotesActivity` in some cases (i.e. large devices in landscape orientation) and `NoteEditorActivity` in some cases (i.e. small devices and large devices in portrait orientation).  Having the `NoteEditorActivity` cast may help someone realize the difference even if the code is refactored and the comments are removed.

Reply:
Great, thanks for explaining the reasoning behind that, it makes sense!

However, bear in mind that static analysis tools will not like that and that it might create confusion for other devs, which like me, might try to verify that everything is okay within a class, traversing it by error/warnings/spelling mistakes and correct anything suspicious. For example, if I where to touch this class at some point in time I would have definitely refactored this code, since it is being greyed out on my IDE and appears as warning.

Another idea to be extra verbose, which also allows getting rid of the comments altogether, is extracting the logic in private methods. I would have done that instead and then cover both cases:
```
if (activity instanceof NotesActivity) {
    showPlaceholderInLandscapeLargeDevices((NotesActivity) activity);
} else if (activity instanceof NoteEditorActivity) {
    closeEditorInPortrait((NoteEditorActivity) activity);
}
```